src/main/scala/uclid/lang/UclidLanguage.scala

```diff
@@ -107,6 +107,21 @@ object ULContext {
     }
     recordTypes.exists(r => r.fields.map(_._1).contains(id))
   }
+  // Leiqi: add checkField
+  def checkuninterpretedField(id : Identifier) : Boolean = {
+    val recordTypes = origTypeMap.fwdMap.map(_._2) flatMap {
+      case r : RecordType => Some(r)
```

Comment:
Shouldn't we be filtering out the set of UninterpretedTypes here?
```suggestion
      case r : UninterpretedType => Some(r)
```